charts/spire/charts/spire-server/templates/_helpers.tpl

```diff
@@ -109,7 +109,7 @@ Create the name of the service account to use
 {{- $root := deepCopy . }}
 {{- $tag := (default $root.image.tag $root.image.version) | toString }}
 {{- if eq (len $tag) 0 }}
-{{- $_ := set $root.image "tag" $root.KubeVersion }}
+{{- $_ := set $root.image "tag" (regexReplaceAll "([^\\+]*)\\+?.*" $root.KubeVersion "${1}") }}
```

Comment:
```suggestion
{{- $_ := set $root.image "tag" (regexReplaceAll "\+.*$" $root.KubeVersion "") }}
```

Maybe this works. This matches the end of the string from `+` till the end and then replaces it with an empty string. Assuming this will give a `1.25.8` if `1.25.8+k3s` is the version.

Reply:
I like this version for its simplicity. But it only slices off with a +. Maybe some distro will use -.

The suggestion I made pulls out anything that looks like semver and drops the rest, so it will cut out more of the possible bad strings.